examples/minimal/packages/client-react-external-wallet/src/mud/setupNetwork.ts

```diff
@@ -24,14 +25,20 @@ export async function setupNetwork() {
     startBlock: BigInt(networkConfig.initialBlockNumber),
   });
 
+  const write$ = new Subject<ContractWrite>();
+  let nextWriteId = 0;
+  const onWrite = (write: ContractWrite) =>
+    write$.next({ id: `${write.id}:${nextWriteId++}`, request: write.request, result: write.result });
+
   return {
     worldAddress: networkConfig.worldAddress,
-    mudConfig,
     tables,
     useStore,
     publicClient,
     latestBlock$,
     storedBlockLogs$,
     waitForTransaction,
+    onWrite,
+    write$: write$.asObservable().pipe(share()),
```

Comment:
this data flow feels a little weird now, curious if you have ideas for improving this

Reply:
Is it weird? 🤔 I think the app requires a single `write$` for the dev-tool, which can be used for any wallet clients. Hence, it made sense to me to include this in the `setup`.